src/components/datatable/useTableData.js

```diff
@@ -0,0 +1,262 @@
+                roundFn = numberPrecision(precision)
+            }
+            return {
+                name: toTitleCase(`${type} ${title}`),
+                dataKey: type,
+                roundFn,
+                type: TYPE_NUMBER,
+            }
+        })
+    }
+
+    return [INDEX, NAME, ID, TYPE]
+        .map((field) => defaultFieldsMap()[field])
+        .concat(customFields)
+}
+
+const EMPTY_AGGREGATIONS = {}
+const EMPTY_LAYER = {}
+
+export const useTableData = ({ layer, sortField, sortDirection }) => {
+    const allAggregations = useSelector((state) => state.aggregations)
+    const aggregations = allAggregations[layer.id] || EMPTY_AGGREGATIONS
+
+    const {
+        layer: layerType,
+        aggregationType,
+        legend,
+        styleDataItem,
+        data,
+        dataFilters,
+        headers: layerHeaders,
+        serverCluster,
+    } = layer || EMPTY_LAYER
+
+    const dataWithAggregations = useMemo(
+        () =>
+            data
+                .map((d, i) => ({
+                    index: i,
+                    ...d,
+                }))
+                .filter((d) => !d.properties.hasAdditionalGeometry)
+                .map((d, i) => ({
+                    ...(d.properties || d),
+                    ...aggregations[d.id],
+                    index: d.index,
+                    i,
+                })),
+        [data, aggregations]
+    )
+
+    const headers = useMemo(() => {
+        if (!layerType || !dataWithAggregations.length) {
+            return []
+        }
+
+        switch (layerType) {
+            case THEMATIC_LAYER:
+                return getThematicHeaders()
+            case EVENT_LAYER:
+                return getEventHeaders({ layerHeaders, styleDataItem })
+            case ORG_UNIT_LAYER:
+                return getOrgUnitHeaders()
+            case EARTH_ENGINE_LAYER:
+                return getEarthEngineHeaders({
+                    aggregationType,
+                    legend,
+                    data: dataWithAggregations,
+                })
+            case FACILITY_LAYER:
+                return getFacilityHeaders()
+            default:
+                // TODO - throw error?
+                return []
+        }
+    }, [
+        layerType,
+        aggregationType,
+        legend,
+        styleDataItem,
+        dataWithAggregations,
+        layerHeaders,
+    ])
+
+    const rows = useMemo(() => {
+        if (!dataWithAggregations.length || !headers?.length) {
+            return []
+        }
+
+        const filteredData = filterData(dataWithAggregations, dataFilters)
+
+        //sort
+        filteredData.sort((a, b) => {
+            a = a[sortField]
+            b = b[sortField]
+
+            if (typeof a === TYPE_NUMBER) {
+                return sortDirection === ASCENDING ? a - b : b - a
+            }
+            // TODO: Make sure sorting works across different locales - use lib method
+            if (a !== undefined) {
```

Comment:
IMO the code would be a bit easier to read if the case of `sortFieldA` being `undefined` is handled first, before the handling of the number type:
```js
if (typeof sortFieldA === 'undefined') {
    return 0
}
```

And then the ternary can be the default return statement.

Reply:
Agreed that we'll look at this in a separate refactor.